include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8904,6 +8904,11 @@ def err_bounds_type_annotation_lost_checking : Error<
 	"%select{assignment|increment|decrement|call|volatile}0 expression not allowed in "
 	"%select{expression|dynamic check expression|count expression|byte count expression|bounds expression}1">;
 
+  def err_checked_scope_type : Error<
+    "invalid %select{parameter type|return type|variable type|member type}0 in checked scope or function">;
+
+  def err_checked_scope_no_prototype_func : Error<
```

Comment:
Yes, I agree that the err_checked_scope_type need should be revised.   I think it would be useful to say something like "parameter|return|variable|member cannot have an unchecked pointer|array type in a checked scope or function"